Content.Server/GameObjects/EntitySystems/AI/Steering/AiSteeringSystem.cs

```diff
@@ -135,7 +135,32 @@ public void Unregister(IEntity entity)
             
             if (_pathfindingRequests.TryGetValue(entity, out var request))
             {
-                request.Item1.Cancel();
+                switch (request.Item2.Status)
+                {
+                    case JobStatus.Pending:
+                        break;
+                    case JobStatus.Running:
+                        request.Item1.Cancel();
+                        break;
+                    case JobStatus.Paused:
+                        request.Item1.Cancel();
+                        break;
+                    case JobStatus.Waiting:
+                        request.Item1.Cancel();
+                        break;
+                    case JobStatus.Finished:
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
```

Comment:
```suggestion
                switch (request.Item2.Status)
                {
                    case JobStatus.Pending:
                    case JobStatus.Finished:
                        break;
                    case JobStatus.Running:
                    case JobStatus.Paused:
                    case JobStatus.Waiting:
                        request.Item1.Cancel();
                        break;
                }
```

Reply:
Also, we have C# 7.0, so these request tuples should really be named.

```cs
private readonly Dictionary<IEntity, (CancellationTokenSource, Job<Queue<TileRef>>)> _pathfindingRequests = 
    new Dictionary<IEntity, (CancellationTokenSource, Job<Queue<TileRef>>)>();
```
Should become
```cs
private readonly Dictionary<IEntity, (CancellationTokenSource CancelToken, Job<Queue<TileRef>> job)> _pathfindingRequests = 
    new Dictionary<IEntity, (CancellationTokenSource CancelToken, Job<Queue<TileRef>> job)>();
```

Which would let you do `request.CancelToken.cancel()` and `request.Job.Status`, I'm sure you can agree that's much more readable!

Reply:
TIL, thanks remie. I definitely needed that for my next AI PR...